### PR TITLE
BFD-3156: Updates BFD Pom Entrypoints

### PR DESCRIPTION
**Pull request by aadedejifearless**
**JIRA Ticket:**
[BFD-3156](https://jira.cms.gov/browse/BFD-3156)

**User Story or Bug Summary:**

As a BFD contributor, I want pom.xml files to serve as both an accurate reference as well as instructions for BFD maven projects so that I can easily learn about the project's technical requirements and continue to build BFD.

---

### What Does This PR Do?

- Updates the `bfd-parent` pom.xml based on the AC. 
- bfd-parent
   - update issueManagement, ciManagement reference URLs
   - remove outdated comments on project.build.outputTimestamp
   - remove local.pac.enabled property

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] The data dictionary has been updated with any field mapping changes, if any were made.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/pom.xml [hunk 1]
```diff
@@ -65,11 +65,7 @@
         <!-- The version of Java that we'll use for everything. -->
         <javaSourceVersion>21</javaSourceVersion>
 
-        <!-- Create a standard (but fake) timestamp for use in creating build artifacts so that
-            they are reproducible bit-by-bit no matter when/where they are generated.
-            See https://maven.apache.org/guides/mini/guide-reproducible-builds.html -->
-
-        <project.build.outputTimestamp>2024-01-10T17:15:29Z</project.build.outputTimestamp>
+        <project.build.outputTimestamp>2024-01-03T21:29:01Z</project.build.outputTimestamp>
```
Comment (lsmitchell) [inline apps/pom.xml:68]: I suppose it probably doesnt matter _when_ this timestamp is, but why was this changed slightly?

**Review by lsmitchell — COMMENTED** [review]
I suppose dsl-codegen-plugin's project version couldnt be removed then? Just asking since it was part of the AC.

File: apps/pom.xml (hunk 1 above)
Comment (aadedejifearless) [reply apps/pom.xml:68]: Dont think it matters. But i'll change it back as that wasn't in the AC. The AC was to remove the outdated comment.

**Comment by aadedejifearless** [comment]
> I suppose dsl-codegen-plugin's project version couldnt be removed then? Just asking since it was part of the AC.

Correct. That wasn't viable.

**Review by lsmitchell — APPROVED** [review]

**Review by mjburling — APPROVED** [review]

**Review by brandoncruz3 — APPROVED** [review]
